.github/workflows/pr_basic_checks.yml

```diff
@@ -0,0 +1,243 @@
+        run: |
+          set -x
+          ls
+          mkdir -p SCANCODE
+          git diff --name-only --diff-filter=ad origin/${GITHUB_BASE_REF}.. \
+            | ( grep '.\(c\|cpp\|h\|hpp\|py\)$' || true ) \
+            | ( grep -v '^tools/test/toolchains/api_test.py' || true ) \
+            | while read file; do cp --parents "${file}" SCANCODE; done
+          ls SCANCODE
+          scancode -l --json-pp scancode.json SCANCODE
+          python ./tools/test/travis-ci/scancode-evaluate.py scancode.json || true
+          cat scancode-evaluate.log
+          COUNT=$(cat scancode-evaluate.log | grep 'File:' | grep -v 'SPDX' | wc -l) || true
+          if [ $COUNT == 0 ]; then
+            echo "License check OK";
+            true;
+          else
+            echo "License check failed, please review license issues found in modified files";
+            false;
+          fi
+
+      - 
+        name: license check new files
+        shell: bash
+        run: |
+          mkdir -p SCANCODE_NEW_FILES
+          git diff --name-only --diff-filter=A origin/${GITHUB_BASE_REF}.. \
+            | ( grep '.\(c\|cpp\|h\|hpp\|py\)$' || true ) \
+            | ( grep -v '^tools/test/toolchains/api_test.py' || true ) \
+            | while read file; do cp --parents "${file}" SCANCODE_NEW_FILES; done
+          ls SCANCODE_NEW_FILES
+          scancode -l --json-pp scancode.json SCANCODE_NEW_FILES
+          python ./tools/test/travis-ci/scancode-evaluate.py scancode.json || true
+          cat scancode-evaluate.log
+          COUNT=$(cat scancode-evaluate.log | grep 'File:' | grep -v 'SPDX' | wc -l) || true
+          if [ $COUNT == 0 ]; then
+            echo "License check OK";
+            true;
+          else
+            echo "License check failed, please review license issues found in new files";
+            false;
+          fi
+
+      - 
+        name: "UTF-8 Check"
+        run: |
+          # Make sure we're not introducing any text which is not UTF-8 encoded
+          git diff origin/${GITHUB_BASE_REF} -U0 | ( grep -a '^+' || true ) | ( ! grep -axv '.*' )
+
+      - 
+        name: "include check"
+        run: |
+          ! git grep '^#include\s["'"']mbed.h['"'"]$' -- '*.c' '*.h' '*.cpp' '*.hpp' \
+              ':!*platform_mbed.h' ':!*TESTS/*' ':!TEST_APPS/' ':!UNITTESTS/' \
+              ':!*tests/*' ':!*targets/*' ':!*TARGET_*' ':!*unsupported/*' \
+              ':!*events/tests/*' ':!*drivers/tests/*'
+
+      - 
+        name: astyle checks
+        run: |
+          git diff --name-only --diff-filter=d origin/${GITHUB_BASE_REF} \
+            | ( grep '.*\.\(c\|cpp\|h\|hpp\)$' || true ) \
+            | ( grep -v -f .codecheckignore || true ) \
+            | while read file; do astyle -n --options=.astylerc "${file}"; done
+          git diff --exit-code --diff-filter=d --color
+
+      - 
+        name: spell checks
+        run: |
+          ./tools/test/travis-ci/doxy-spellchecker/spell.sh drivers .codecheckignore
```

Comment:
There are many places referencing `./tools/test/travis-ci`
Shall we rename these as well ?

Reply:
I thought about it. But I am not sure if these are referenced somewhere else. May be we should risk and rename it?